pyscriptjs/src/components/pyscript.ts

```diff
@@ -3,19 +3,26 @@ import {
     addPostInitializer,
     addToScriptsQueue,
     loadedEnvironments,
-    pyodideLoaded,
+    runtimeLoaded,
     type Environment,
 } from '../stores';
+
+/*
+All references to `PyodideInterface` have been replaced with
+`Runtime` which has the available methods of running code asynchronously
+etc. which each runtime is responsible for implementing on its own.
+*/
+
 import { addClasses, htmlDecode } from '../utils';
 import { BaseEvalElement } from './base';
-import type { PyodideInterface } from '../pyodide';
+import type { Runtime } from '../runtime';
 
-// Premise used to connect to the first available pyodide interpreter
-let pyodideReadyPromise: PyodideInterface;
+// Premise used to connect to the first available runtime (can be pyodide or others)
+let runtimeReadyPromise: Runtime;
 let environments: Record<Environment['id'], Environment> = {};
 
-pyodideLoaded.subscribe(value => {
-    pyodideReadyPromise = value;
+runtimeLoaded.subscribe(value => {
+    runtimeReadyPromise = value;
```

Comment:
I think we should call it runtime. It's not a promise anymore.

Reply:
nice nitpick 🗡️